src/main/java/com/commercetools/pingpong/model/Game.java

```diff
@@ -54,38 +54,41 @@ public void updateScoreOfPlayer(Player player, int actionType) {
     }
 
     public void updateMatchScoreOfPlayers() {
+        if (hasPlayerWonSet(playerOne, playerTwo) && !isMatchOver()) {
+            this.playerOne.addMatchScore();
+            this.playerOne.reset();
+            this.playerTwo.reset();
 
+        } else if (hasPlayerWonSet(playerTwo, playerOne) && !isMatchOver()) {
+            this.playerTwo.addMatchScore();
+            this.playerOne.reset();
+            this.playerTwo.reset();
 
-
-     if (hasPlayerWonSet(playerOne, playerTwo) && !isMatchOver()) {
-         this.playerOne.addMatchScore();
-         this.playerOne.reset();
-         this.playerTwo.reset();
-
-     } else if (hasPlayerWonSet(playerTwo, playerOne) && !isMatchOver()) {
-         this.playerTwo.addMatchScore();
-         this.playerOne.reset();
-         this.playerTwo.reset();
-
-     }
-     if (this.playerOne.getMatchScore() == 2) {
-            System.out.println("The match is Over! Player1 has won!!!");
+        }
+        if (this.playerOne.getMatchScore() == 2) {
+            //System.out.println("The match is Over! Player1 has won!!!");
             this.playerOne.reset();
             this.playerTwo.reset();
             this.playerOne.resetMatchScore();
```

Comment:
It seems that reset() method also resets the match score, so maybe the `resetMatchScore()` calls here are redundant ;)